components/compositions/layer-parser/layer-parser.service.ts

```diff
@@ -0,0 +1,401 @@
+import '../../add-layers/vector/add-layers-vector.module';
+import ImageLayer from 'ol/layer/Image';
+import SparqlJson from '../../layers/hs.source.SparqlJson';
+import VectorLayer from 'ol/layer/Vector';
+import {Attribution} from 'ol/control';
+import {HsAddLayersVectorService} from '../../add-layers/vector/add-layers-vector.service';
+import {HsMapService} from '../../map/map.service';
+import {HsStylerService} from '../../styles/styler.service';
+import {ImageArcGISRest, ImageStatic, TileArcGISRest, TileWMS} from 'ol/source';
+import {ImageWMS, XYZ} from 'ol/source';
+import {Injectable} from '@angular/core';
+import {Tile} from 'ol/layer';
+import {Vector as VectorSource} from 'ol/source';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class HsCompositionsLayerParserService {
+  constructor(
+    private HsMapService: HsMapService,
+    private HsAddLayersVectorService: HsAddLayersVectorService,
+    private HsStylerService: HsStylerService
+  ) {}
+
+  /**
+   * @ngdoc method
+   * @name hs.compositions.config_parsers.service#createWmsLayer
+   * @public
+   * @param {object} lyr_def Layer definition object
+   * @returns {object} Ol Image or Tile layer
+   * @description Parse definition object to create WMS Ol.layer  (source = ol.source.ImageWMS / ol.source.TileWMS)
+   */
+  createWmsLayer(lyr_def) {
+    const source_class = lyr_def.singleTile ? ImageWMS : TileWMS;
+    const layer_class = lyr_def.singleTile ? ImageLayer : Tile;
+    const params = lyr_def.params;
+    const legends = [];
+    delete params.REQUEST;
+    //delete params.FORMAT; Commented, because otherwise when loading from cookie or store, it displays jpeg
+    if (lyr_def.legends) {
+      for (let idx_leg = 0; idx_leg < lyr_def.legends.length; idx_leg++) {
```

Comment:
This seems like a horribly unreadable variant of for..of loop to me. If I am correct, I propose refactoring it.